Classes/Imgur.m

```diff
@@ -0,0 +1,196 @@
+//
+//  Imgur.m
+//  LatestChatty2
+//
+//  Created by Boarder2 on 3/7/2020.
+//
+
+#import "Imgur.h"
+
+@implementation Imgur
+
+@synthesize image, delegate;
+
+- (id)initWithImage:(UIImage *)anImage {
+	if (!(self = [super init])) return nil;
+	self.image = anImage;
+	return self;
+}
+
+- (NSData *)compressJpeg:(CGFloat)quality {
+	return UIImageJPEGRepresentation(image, quality);
+}
+
+- (void)informDelegateOnMainThreadWithURL:(NSString*)url {
+	if (url) {
+        [delegate image:self sendComplete:url];
+    } else {
+        [delegate image:self sendFailure:@"stub"];
+    }
+}
+
+- (void)uploadAndReturnImageUrlWithDictionary:(NSDictionary*)args {
+    // Use the AFNetworking to login and stream image data
+    AFHTTPSessionManager *manager = [AFHTTPSessionManager manager];
+    manager.requestSerializer = [AFHTTPRequestSerializer serializer];
+    manager.responseSerializer = [AFJSONResponseSerializer serializer];
+
+	NSData *imageData = [self compressJpeg:[args[@"qualityAmount"] floatValue]];
+	NSMutableURLRequest *request = [[AFHTTPRequestSerializer serializer]
+			multipartFormRequestWithMethod:@"POST"
+								 URLString:@"https://api.imgur.com/3/image"
+								parameters:nil
+				 constructingBodyWithBlock:^(id<AFMultipartFormData> formData) {
+					 [formData appendPartWithFileData:imageData name:@"image" fileName:@"iPhoneUpload.jpg" mimeType:@"image/jpeg"];
+				 } error:nil];
+	[request addValue:@"Client-ID REPLACEME" forHTTPHeaderField:@"Authorization"];
```

Comment:
This needs to have a client ID set. Probably want to do this during build time.

https://api.imgur.com/oauth2/addclient